### ISSUE TITLE
Greek font looks like shit/ rendering issue 

### ISSUE THREAD

**Issue by stimfrog** [duolicious/duolicious-frontend#507]
![Screenshot 2024-12-17 at 21-59-15 (1) Duolicious](https://github.com/user-attachments/assets/18e86696-0a5a-4f2e-84b6-1f393c3f77d5)

**Comment by kero3009destiny** [comment]
Inter or Roboto can be a good fallback for Greek, pretty close to Montserrat.

**Comment by DeepeshKalura** [comment]
@duogenesis assigned this task to me.

We have to do simple task.

1. I will download inter and roboto first.  then put in the assets folder.
2. There must a global theme file I will put that as a fallback where I found the greek. 
3. I think it just need to do in one are and valla the issue is resolve I think.

**Comment by duogenesis** [comment]
@DeepeshKalura Fonts go in the `assets/` directory. Fonts are loaded in `App.tsx`. You can replace Montserrat with Roboto entirely, Roboto's license permits us to distribute the font with Duolicious.

**Comment by DeepeshKalura** [comment]
@duogenesis  
Yeah thanks man but i predicted that. Since, you are using Montserrat, which is a Google Font, and Roboto is also a Google Font, I assume there shouldn't be any issues with the licensing. 

However, I recommend verifying this information here: https://fonts.google.com/specimen/Roboto better then sorry

**Comment by duogenesis** [comment]
@DeepeshKalura License looks fine to me

**Comment by DeepeshKalura** [comment]
> @DeepeshKalura Fonts go in the `assets/` directory. Fonts are loaded in `App.tsx`. You can replace Montserrat with Roboto entirely, if Roboto's license permits us to distribute the font with Duolicious.

Hey man replace Montserrat with roboto entirely mean? 

I was thinking to add new fonts roboto loadFonts and then in default text component ( might is the component ) make a roboto default and fallback montserrat.  

I don't why we are removing Montserrat. I don't know where this image from by stimfrog  from the codebase help me in that also

**Comment by duogenesis** [comment]
>Hey man replace Montserrat with roboto entirely mean?

Yes

>I was thinking to add new fonts roboto loadFonts and then in default text component ( might is the component ) make a roboto default and fallback montserrat.

That sounds right, except for the fallback

>I don't why we are removing Montserrat

If we use Roboto, it'll do everything Montserrat does but better. So Montserrat won't be needed

>I don't know where this image from by stimfrog from the codebase

It's not in the code base. It looks like he wrote some Greek text in the 'About' section of his Profile. By the way, I forgot to mention you'll need to update the font in `components/default-text-input.tsx`.

**Comment by DeepeshKalura** [comment]
@duogenesis  Ok, so I am mostly correct for this time I will not be removing anything you will tag me in the code review stuff.  Then I will start removing it if everything seems fine to you. 

Rather I will create entirely new things separately and you check first tell me how you check also I think I will do it within 1 hour.

so I also change the `component/default-text-input.tsx` change ok

**Comment by duogenesis** [comment]
@DeepeshKalura Are you going to fork [duolicious/duolicious-frontend](https://github.com/duolicious/duolicious-frontend) and then raise a pull request which contains your changes?

**Comment by DeepeshKalura** [comment]
@duogenesis  i already did and did some changes and reading codebase. LOL it first time seeing react-native code by the I am do mostly flutter for frontend

**Comment by duogenesis** [comment]
I've looked into this some more. Can't find any fonts I like which support a wide variety of commonly used languages. For example, I found some fonts which support Greek and English, but then they don't support Japanese.

Fallback fonts aren't an option as React Native doesn't support them, and probably won't any time soon because of how infrequent the requests have been over the past few years  (facebook/react-native#48625, facebook/react-native#32396, facebook/react-native#25798). Crappy fallback support can be hacked together by using logic to detect languages and dynamically changing the font, though that approach will break if someone uses two languages in the same bit of text.

If Facebook ever implements font fallbacks, I might replace [Montserrat](https://fonts.google.com/specimen/Montserrat?preview.text=Online%20dating,%20but%20based%20and%20true%20love-pilled%20%F0%9F%92%95&preview.size=15) with [Geologica](https://fonts.google.com/specimen/Geologica?preview.text=Online%20dating,%20but%20based%20and%20true%20love-pilled%20%F0%9F%92%95&preview.size=15&categoryFilters=Sans+Serif:%2FSans%2F*), [Fira Sans](https://fonts.google.com/specimen/Fira+Sans?preview.text=Online%20dating,%20but%20based%20and%20true%20love-pilled%20%F0%9F%92%95&preview.size=15&categoryFilters=Feeling:%2FExpressive%2FBusiness;Sans+Serif:%2FSans%2F*), or [Commissioner](https://fonts.google.com/specimen/Commissioner?preview.text=Online%20dating,%20but%20based%20and%20true%20love-pilled%20%F0%9F%92%95&preview.size=15&categoryFilters=Feeling:%2FExpressive%2FBusiness;Sans+Serif:%2FSans%2F*) to cover English and Greek. Then I'll find a fallback for the other languages weebs use.

**Comment by kero3009destiny** [comment]
If that is so, keeping the font as is or switching to one that looks similar but has at least one more supported script may be the safer option. I think duo will eventually need a full re-design at which point fonts can be reconsidered. I "professionally" do UI design and UX research and definitely want to discover and explore what may align with "what duolicious is" without being too corny or overwhelming.

**Comment by duogenesis** [comment]
It would've been nice to have a professional to design the UI/UX before I wrote a line of code. Although I didn't have a good sense of "what Duolicious is" when I first made it, and the app's home page was very generic a couple of years ago. But now it's grown into the "chronically online" niche. So a UI that resonates most with the typical Discord user might be appropriate in the future.
